Sources/Storyboard+Swizzling.m

```diff
@@ -0,0 +1,67 @@
+//
+//  Storyboard+Swizzling.m
+//  SwinjectStoryboard
+//
+//  Created by Mark DiFranco on 2017-03-31.
+//  Copyright © 2017 Swinject Contributors. All rights reserved.
+//
+
+#import "Storyboard+Swizzling.h"
+#import <objc/runtime.h>
+#import "SwinjectStoryboard-Swift.h"
+
+#ifdef os(iOS) || os(tvOS)
+
+typedef UIStoryboard Storyboard;
+
+#elseif os(OSX)
+
+typedef NSStoryboard Storyboard;
+
+#endif
+
+@implementation UIStoryboard (Swizzling)
+
++ (void)load {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        Class class = object_getClass((id)self);
+
+        SEL originalSelector = @selector(storyboardWithName:bundle:);
+        SEL swizzledSelector = @selector(swinject_storyboardWithName:bundle:);
+
+        Method originalMethod = class_getClassMethod(class, originalSelector);
+        Method swizzledMethod = class_getClassMethod(class, swizzledSelector);
+
+        BOOL didAddMethod =
+        class_addMethod(class,
+                        originalSelector,
+                        method_getImplementation(swizzledMethod),
+                        method_getTypeEncoding(swizzledMethod));
+
+        if (didAddMethod) {
+            class_replaceMethod(class,
+                                swizzledSelector,
+                                method_getImplementation(originalMethod),
+                                method_getTypeEncoding(originalMethod));
+        } else {
+            method_exchangeImplementations(originalMethod, swizzledMethod);
+        }
+    });
+}
+
++ (nonnull instancetype)swinject_storyboardWithName:(NSString *)name, bundle:(UIBundle)storyboardBundleOrNil {
+    if (self == [Storyboard class]) {
+        
+        // Instantiate SwinjectStoryboard if UI/NSStoryboard is trying to be instantiated.
+        if [SwinjectStoryboard isCreatingStoryboardReference] {
```

Comment:
Need to `#import "SwinjectStoryboard-Swift.h"` in order to access this Swift class method, but for some reason the header is not automatically generated.